packages/nimble-components/src/table/specs/table-column-specs/table-column-mapping.md

```diff
@@ -0,0 +1,299 @@
+# Mapping Table Column
+
+## Overview
+
+The `nimble-table-column-mapping` is a component that supports rendering specific number, boolean, or string values as mapped icons, spinners, or text. `nimble-table-column-icon` is a specialized version of `nimble-table-column-mapping` which defaults to having a minimal, fixed width. The actual mappings are defined by child elements `nimble-mapping-icon`, `nimble-mapping-spinner`, and `nimble-mapping-text`.
+
+### Background
+
+[Icon column type issue](https://github.com/ni/nimble/issues/1013)
+
+[Boolean column type issue](https://github.com/ni/nimble/issues/1103)
+
+### Features
+
+-   Supported input:
+    -   string
+    -   number
+    -   boolean
+-   Supported output:
+    -   Text
+    -   Icon
+    -   Spinner
+    -   (empty)
+
+### Non-goals
+
+-   Non-Nimble icon support
+-   Arbitrary icon colors
+-   Hyperlink icons
+-   Mixed text and icons
+-   Non-icon, non-spinner Nimble components
+
+---
+
+## Design
+
+Below is an example of how these elements would be used within a `nimble-table`:
+
+```HTML
+<nimble-table>
+    <nimble-table-column-icon field-name="status">
+        Status
+        <nimble-mapping-icon key="fail" icon="nimble-icon-xmark" severity="error" label="Failed"></nimble-mapping-icon>
+        <nimble-mapping-icon key="error" icon="nimble-icon-xmark" severity="error" label="Errored"></nimble-mapping-icon>
+        <nimble-mapping-icon key="pass" icon="nimble-icon-check" severity="success" label="Passed"></nimble-mapping-icon>
+        <nimble-mapping-spinner key="running" label="Running..."></nimble-mapping-spinner>
+    </nimble-table-column-icon>
+    <nimble-table-column-mapping field-name="errorCode" data-type="number">
+        Error Code
+        <nimble-mapping-text key="1" label="A bad thing happened"></nimble-mapping-text>
+        <nimble-mapping-text key="2" label="A worse thing happened"></nimble-mapping-text>
+        <nimble-mapping-text key="3" label="A terrible thing happened"></nimble-mapping-text>
+    </nimble-table-column-mapping>
+    <nimble-table-column-icon field-name="archived" data-type="boolean">
+        Archived
+        <nimble-mapping-icon key="true" icon="nimble-icon-database" label="Archived"></nimble-mapping-icon>
+    </nimble-table-column-icon>
+</nimble-table>
+```
+
+Note that the `key` attribute values are always given as strings. In the case of boolean or number mappings, this value is parsed to the appropriate type, so that it can properly be compared to the values from the row records.
+
+When none of the given mappings match the record value for a cell, that cell will be empty. Alternatively, we could provide support for marking a mapping with a `default` attribute that would cause it to be chosen when no other values matched. This would be equivalent to the `placeholder` configuration we provide on `nimble-table-column-text` and `nimble-table-column-anchor`. I don't think we need to support this initially.
```

Comment:
Is there a reason to not support a placeholder initially for the `nimble-table-column-mapping` column? I could see a client wanting to specify some sort of "default" value if the value doesn't match one of the enumerated `key` values.

Reply:
No real reason. Frankly, I just anticipate it being pretty tricky to implement in a template with just `repeat` and `when`, but maybe I'm wrong. I'll include it in the plan.

Reply:
Turned out not to be that tricky.